src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -43,7 +129,7 @@ internal static bool HasIllegalCharacters(string path, bool checkAdditional = fa
             Contract.Requires(path != null);
 
             // Question mark is a normal part of extended path syntax (\\?\)
-            int startIndex = path.StartsWith(LongPathPrefix, StringComparison.Ordinal) ? LongPathPrefix.Length : 0;
+            int startIndex = PathInternal.IsExtended(path) ? ExtendedPathPrefix.Length : 0;
```

Comment:
Are we now going to be doing many more null checks because each of these helpers does one?  Or do those get optimized away?